tests/ui/use/use-path-segment-kw.rs

```diff
@@ -0,0 +1,177 @@
+
+mod foo {
+    pub mod bar {
+        pub mod foobar {
+            pub mod qux {
+                pub use super::inner;
+            }
+
+            pub mod baz {
+                pub use super::inner;
+            }
+
+            pub fn inner() {}
+        }
+
+        // --- $crate ---
+        macro_dollar_crate!(foobar);
+
+        // --- crate ---
+        use crate; //~ ERROR crate root imports need to be explicitly named: `use crate as name;`
+        pub use crate as _crate; // Good
+
+        use ::crate; //~ ERROR `crate` in paths can only be used in start position
+        use ::crate as _crate2; //~ ERROR `crate` in paths can only be used in start position
+        use ::{crate}; //~ ERROR `crate` in paths can only be used in start position
+        use ::{crate as _nested_crate2}; //~ ERROR `crate` in paths can only be used in start position
+
+        use foobar::crate; //~ ERROR `crate` in paths can only be used in start position
+        use foobar::crate as _crate3; //~ ERROR `crate` in paths can only be used in start position
+        use foobar::{crate}; //~ ERROR `crate` in paths can only be used in start position
+        use foobar::{crate as _nested_crate3}; //~ ERROR `crate` in paths can only be used in start position
+
+        use crate::crate; //~ ERROR `crate` in paths can only be used in start position
+        use crate::crate as _crate4; //~ ERROR `crate` in paths can only be used in start position
+        use crate::{crate}; //~ ERROR `crate` in paths can only be used in start position
+        use crate::{crate as _nested_crate4}; //~ ERROR `crate` in paths can only be used in start position
+
+        use super::crate; //~ ERROR `crate` in paths can only be used in start position
+        use super::crate as _crate5; //~ ERROR `crate` in paths can only be used in start position
+        use super::{crate}; //~ ERROR `crate` in paths can only be used in start position
+        use super::{crate as _nested_crate5}; //~ ERROR `crate` in paths can only be used in start position
+
+        use self::crate; //~ ERROR `crate` in paths can only be used in start position
+        use self::crate as _crate6; //~ ERROR `crate` in paths can only be used in start position
+        use self::{crate}; //~ ERROR `crate` in paths can only be used in start position
+        use self::{crate as _nested_crate6}; //~ ERROR `crate` in paths can only be used in start position
+
+        // --- super ---
+        use super; //~ ERROR imports need to be explicitly named
+        pub use super as _super; // Good
+
+        use ::super; //~ ERROR imports need to be explicitly named
+        use ::super as _super2; //~ ERROR unresolved import `super`
+        use ::{super}; //~ ERROR imports need to be explicitly named
+        use ::{super as _nested_super2}; //~ ERROR unresolved import `super`
+
+        use foobar::super; //~ ERROR imports need to be explicitly named
+        pub use foobar::super as _super3; // Good
```

Comment:
This is not good, `super` is only supposed to be used in start position, or after another `super`.
`super`, `super::super`, `super::super::super` - ok, `foobar::super` - not ok, `self::super` - not ok even if accepted on stable.

It does have a reasonable meaning (technically we can support `self` and `super` in all positions in paths), but relaxing the rules here is probably a different issue than what this PR tries to address.